projects/question/src/main/java/org/batfish/question/jsonpath/JsonPathQuestionPlugin.java

```diff
@@ -159,7 +162,10 @@ public JsonPathAnswerElement answer() {
                 nodePathResult.setPath(nodesPath);
                 nodePathResult.setNumResults(numResults);
                 boolean includeSuffix = nodesPath.getSuffix();
-                if (!nodesPath.getSummary()) {
+                if (nodesPath.hasValidAssertion()) {
+                  boolean assertion = nodesPath.getAssertion().evaluate(suffixes);
+                  nodePathResult.setAssertionResult(assertion);
+                } else if (!nodesPath.getSummary()) {
```

Comment:
It's not about expanding the operators. Right now, the assertion operates only on the suffix (rather than, say, the entire JsonPath result) -- it's not possible to extend cleanly to including the prefix as well, not without adding something like prefixAssertion or wholeResultAssertion, e.g., – different operators entirely.